src/components/new-request/search.vue

```diff
@@ -250,7 +250,7 @@
         <!-- federal sub-flow -->
         <XproFederalBullets v-if="isFederal && getSearchBusiness" />
 
-        <template v-if="(isRestorable && !isFederal) && (isNamedCompany || !isSupportedRestoration(getEntityTypeCd))">
+        <template v-if="(isRestorable && !isFederal) && (isNamedCompany || isCooperative || isCreditUnion || isSelectedCompanyXPro)">
```

Comment:
add it to make sure Extrapos, Coop, and Credit Union can enter their name
![image](https://github.com/bcgov/namerequest/assets/144159934/340eb554-d09c-4e68-b668-6aa76efd56bd)
![image](https://github.com/bcgov/namerequest/assets/144159934/9c302b6a-b73a-4e33-82f0-d840c00278a2)
![image](https://github.com/bcgov/namerequest/assets/144159934/429f0d7d-a471-4fe5-b99b-387e9ce9dbf7)

Reply:
> add it to make sure Extrapos, Coop, and Credit Union can enter their name

The existing getter `isNamedCompany` should be True for these already, so you shouldn't need to check for them explicitly.

In fact, you shouldn't check for entity types here at all. The purpose of the lists in request-action-mappings.ts is to declare which types are valid for which filings. The getter that uses those lists is `isRestorable`, which is already checked in your if statement here. So, the logic here needs to be cleaned up.

Finally, if you see a lot of logic like this in the template then it should be extracted into a local getter (computed), so that the template is as clean as possible.

Reply:
> > add it to make sure Extrapos, Coop, and Credit Union can enter their name
> 
> The existing getter `isNamedCompany` should be True for these already, so you shouldn't need to check for them explicitly.
> 
> In fact, you shouldn't check for entity types here at all. The purpose of the lists in request-action-mappings.ts is to declare which types are valid for which filings. The getter that uses those lists is `isRestorable`, which is already checked in your if statement here. So, the logic here needs to be cleaned up.
> 
> Finally, if you see a lot of logic like this in the template then it should be extracted into a local getter (computed), so that the template is as clean as possible.

I find that if I only keep `isNamedCompany`, it does not work for Coop and Xpro, no name input
![image](https://github.com/bcgov/namerequest/assets/144159934/b8584157-77ef-432b-8489-73f21b5161ec)

Reply:
I notice that if I only keep `isNamedCompany`, it doesn't work for Coop and XPro as no place for name input
![image](https://github.com/bcgov/namerequest/assets/144159934/2cb903c4-c1df-495b-83d2-9845dbe5f3d9)

So maybe I should great a getter to nest all the logic at here?

Reply:
Re: `(isRestorable && !isFederal) && (isNamedCompany || isCooperative || isCreditUnion || isSelectedCompanyXPro)`

I worked through this logic and I think you are correct. The issue is that CP and FI are special cases -- they are handled differently from the top and bottom group here:

![image](https://github.com/bcgov/namerequest/assets/10002889/90f55a06-ef11-4d12-ac66-b7a7440bd798)

So, your code needs to allow them, as you've done 👍 

However, I am not sure about `isSelectedCompanyXPro`. That getter uses `isSelectedXproAndRestorable`, which is already used by `isRestorable`, so it feels like you're checking the same thing twice. Can you explain why you need it? An example would be great.

Reply:
I keep `isSelectedCompanyXPro` because the XPro company should be in different logic, and I didn't find a function that can only check whether it's XPro and avoid `isRestorable`. So should I create a new getter that check XPro only or ?

If I remove `isSelectedCompanyXPro`, it will show, no name input here
![image](https://github.com/bcgov/namerequest/assets/144159934/6f680234-3ba3-4d8e-ac84-c2cb0af60712)

Reply:
All right. The logic is complicated because of the different types of checks. Let's move this to a local getter (computed) where you can explicitly label the different conditions and explain things for future developers.

See following example. This is essentially what you have but with comments. Also I'm using `isSelectedXproAndRestorable` instead of `isSelectedCompanyXPro` (and I think you can delete `isSelectedCompanyXPro` as it's not used for anything else).

```
const isPromptNameInput = (
  this.isNamedCompany || // from radio buttons
  this.isCooperative || // specially include CP
  this.isCreditUnion || // specially include FI
  this.isSelectedXproAndRestorable // include all xpros
)

return (
  this.isRestorable && // check mapping lists for REH
  isPromptNameInput &&
  !this.isFederal // ignore Federal xpros (special case)
)
```

Reply:
Shaoyun, in the end I think you're correct, but I had to wrap my head around it. It really seemed like some of these checks were unnecessary or redundant, but they're checking either types or different parts of the flow. It's complicated. In these cases, I like to see explicit code, comments, etc. You can rewrite the above however you see fit. Thanks for your patience with me :)

Reply:
Sure, Severin, I will create a local getter. Yes, there are lots of complicated logic so I just push lots of comments to fix the errors. And thank you so much for those reviews<

Reply:
> const isPromptNameInput = (
>   this.isNamedCompany || // from radio buttons
>   this.isCooperative || // specially include CP
>   this.isCreditUnion || // specially include FI
>   this.isSelectedXproAndRestorable // include all xpros
> )
> 
> return (
>   this.isRestorable && // check mapping lists for REH
>   isPromptNameInput &&
>   !this.isFederal // ignore Federal xpros (special case)
> )

Hi Severin, should I keep those comments in my getter function or just leave one overall comment for the getter?

Reply:
I don't like to tell smart developers how to code, so up to you where the comments go or exactly what they say but I think they're all valuable. (They are the things I had to read into and think about.)

Reply:
Resolving this conversation as it is now continued elsewhere.